src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,159 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int quantity = 0;
+    private Node<T> node;
+    private Node<T> temp;
+
     @Override
     public void add(T value) {
+        rewindNodeToHead();
+        if (quantity == 0) {
+            node = new Node(null, value, null, true, true);
+        } else {
+            rewindToLastNode();
+            node.setTail(false);
```

Comment:
Don't use getters and setters in class Node. You have direct access to the fields of the inner class Node, so using setters like setTail is unnecessary.